wf-galleon/EAP7-1218_Galleon_Maven_Repository.adoc

```diff
@@ -0,0 +1,74 @@
+= EAP7-1218: Provide a maven repo zip with everything needed to provision EAP using Galleon
+:author:            Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Offer the ability to package all Galleon required artifacts to provision WildFly server.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1218[EAP7-1218]
+
+* https://issues.jboss.org/browse/WFGP-132[WFGP-132]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-891[EAP7-891]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* OpenShift
+
+=== Other Interested Projects
+
+* WildFly Galleon
+
+== Requirements
+
+In order to address this requirement, we are defining a new WildFly Maven plugin (_org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin_)
+goal named _generate-maven-repository_.
+This goal generates a zipped maven repository from a feature-pack.
+
+When building a WildFly release (-Drelease=true), this plugin will be called to 
+generate /target/wildfly-<version>-galleon-maven-repository.zip file.
+
+This zip file should then be made available from WildFly public releases web site (along with released WildFly).
```

Comment:
Ah, so this doesn't end up being sent to the remote maven repo as part of the normal mvn deploy process?

That's a concern as the remote maven repo acts as a kind of "source of truth" for what goes into wildfly.org/downloads.  We populate the folder that gets served by pulling down the needed files from the maven repo. The person doing that task need not be the person who did the mvn deploy.

Reply:
@bstansberry , I am not familiar with the deployment process. I suppose we could attach the zipped repository as a maven artifact to deploy. When building dist with -Drelease, I can see wildfly-dist-17.0.0.Beta1-SNAPSHOT.tar.gz being installed in local repo but not wildfly-17.0.0.Beta1-SNAPSHOT.tar.gz that is downloadable from https://download.jboss.org/wildfly/
Is it renamed when pulling it down to wildfly.org folder?

Reply:
@bstansberry, I made the zipped repository a proper deployed artifact. The proper way to do it is:
1) Change the Galleon plugin to output a repository directory
2) assembly is zipping the directory
3) install plugin is installing it as :
wildfly-dist-17.0.0.Beta1-SNAPSHOT-galleon-maven-repository.zip

Reply:
Sounds good.

And yes, when putting the zips, tars etc in the dir from which https://download.jboss.org/wildfly/ serves, the file name gets tweaked. I think that's likely for historical reasons.